sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -221,4 +221,40 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
       emptyTableData.agg(sumDistinct('a)),
       Row(null))
   }
+
+  test("moments") {
+    checkAnswer(
+      testData2.agg(skewness('a)),
+      Row(0.0))
+
+    checkAnswer(
```

Comment:
So there are a few options for how to go about this. In spark-testing-base (which I'd like to eventually merge some of the functionality of) I made a modified version of checkAnswer which also takes tolerance and does tolerance based matching for floating point data (see https://github.com/holdenk/spark-testing-base/blob/master/src/main/scala/com/holdenkarau/spark/testing/DataFrameSuiteBase.scala & approxEqualDataFrames + approxEquals for comparing the rows). The other option would be, since for these tests its just a single value, just collect the result back, extract the value and do a regular assertion with tolerance. 

I think collecting the result and just checking the single value is probably the best plan for this PR and eventually maybe looking at a more general version of check answer.